astroquery/sha/core.py

```diff
@@ -145,7 +146,9 @@ def parse_line(line, cs=cs):
     # To table
     # transpose data for appropriate table instance handling
     t = Table(list(zip(*data)), names=col_names, dtype=dtypes)
-    return t
+    if len(data) != len(col_names):
+        warnings.warn(NoResultsWarning("No matching rows were found in the query."))
+        return t
```

Comment:
The warning is great, but in the no results case we want to return an empty Table, not `t`, and if we aren't in the no results case we do want to return `t`.